tests/unit/lastuser_core/test_model_user_User.py

```diff
@@ -0,0 +1,466 @@
+        self.assertIsInstance(lena, models.User)
+        self.assertEqual(user.username, u"lena")
+        self.assertEqual(user.fullname, u"Lena Audrey Dachshund")
+
+    def test_user_is_valid_username(self):
+        """
+        Test to check if given is a valid username associated with the user
+        """
+        crusoe = self.fixtures.crusoe
+        # scenario 1: not a valid username
+        number_one = models.User(username=u'number1', fullname=u'Number One')
+        self.assertFalse(number_one.is_valid_username(u'Number1'))
+        # scenario 2: a valid username but not the username of instance passed
+        self.assertFalse(crusoe.is_valid_username(u"oakley"))
+        # scenario 3: a existing username
+        self.assertTrue(crusoe.is_valid_username(u"crusoe"))
+        # scenario 4: a existing org
+        batdog = self.fixtures.batdog
+        self.assertFalse(crusoe.is_valid_username(batdog.name))
+
+    def test_user_profileid(self):
+        """
+        Test to verify profileid (same as username) if any
+        """
+        crusoe = self.fixtures.crusoe
+        # scenario 1: when username is given
+        self.assertEqual(crusoe.profileid(), crusoe.username)
+        # scenario 2: when username doesn't exist
+        mollie = models.User(fullname=u'Mollie')
+        self.assertEqual(len(mollie.profileid()), 22)
+
+    def test_user_displayname(self):
+        """
+        Test to verify any fullname or username or userid for displayname
+        """
+        crusoe = self.fixtures.crusoe
+        oakley = self.fixtures.oakley
+        self.assertEqual(crusoe.displayname(), crusoe.fullname)
+        self.assertEqual(oakley.displayname(), oakley.username)
+        lena = models.User()
+        db.session.add(lena)
+        db.session.commit()
+        self.assertEqual(lena.displayname(), lena.userid)
+
+    def test_user_pickername(self):
+        """
+        Test to verify fullname and username (if any)
+        """
+        # scenario 1: when username exists
+        crusoe = self.fixtures.crusoe
+        result = crusoe.pickername
+        expected_result = u'{fullname} (~{username})'.format(fullname=crusoe.fullname, username=crusoe.username)
+        self.assertEqual(result, expected_result)
+        # scenario 2: when username doesnt exist
+        mr_fedrick = models.User(fullname=u'Mr. Fedrick')
+        result = mr_fedrick.pickername
+        expected_result = u'{fullname}'.format(fullname=mr_fedrick.fullname)
+        self.assertEqual(result, expected_result)
+
+    def test_user_is_profile_complete(self):
+        """
+        Test to check if user profile is complete that is fullname, username
+        and email are present
+        """
+        crusoe = self.fixtures.crusoe
+        self.assertTrue(crusoe.is_profile_complete)
+        lena = models.User()
+        db.session.add(lena)
+        db.session.commit()
+        self.assertFalse(lena.is_profile_complete())
```

Comment:
Shouldn't you be testing the intermediate test cases as well?

`is_profile_complete()` should be false when
 - There is a fullname but no username and email
 - There is a username but no fullname and email
 - There is an email but no fullname and username
 - etc.